src/alpaca_parser.yrl

```diff
@@ -449,11 +452,20 @@ tuple -> '(' tuple_list ')' :
 
 infix -> term op term : make_infix('$2', '$1', '$3').
 
+literal_fun -> fn terms '->' simple_expr:
```

Comment:
Good point, thanks.  We can re-use `validate_args` in the parser here with a small change to reject `#alpaca_fun{}` as a valid argument.